stacker/providers/aws/default.py

```diff
@@ -362,30 +363,63 @@ def create_change_set(cfn_client, fqn, template, parameters, tags,
     return changes, change_set_id
 
 
+def check_tags_present(actual, expected):
+    actual_set = set((item["Key"], item["Value"]) for item in actual)
+    expected_set = set((item["Key"], item["Value"]) for item in expected)
+
+    return actual_set >= expected_set
+
+
 class Provider(BaseProvider):
 
     """AWS CloudFormation Provider"""
 
     DELETED_STATUS = "DELETE_COMPLETE"
+
     IN_PROGRESS_STATUSES = (
         "CREATE_IN_PROGRESS",
-        "UPDATE_COMPLETE_CLEANUP_IN_PROGRESS",
         "UPDATE_IN_PROGRESS",
         "DELETE_IN_PROGRESS",
+        "UPDATE_COMPLETE_CLEANUP_IN_PROGRESS",
+    )
+
+    ROLLING_BACK_STATUSES = (
+        "ROLLBACK_IN_PROGRESS",
+        "UPDATE_ROLLBACK_IN_PROGRESS"
+    )
+
+    FAILED_STATUSES = (
+        "CREATE_FAILED",
+        "ROLLBACK_FAILED",
+        "ROLLBACK_COMPLETE",
+        "DELETE_FAILED",
+        "UPDATE_ROLLBACK_FAILED",
+        "UPDATE_ROLLBACK_COMPLETE",
```

Comment:
This is in both Complete & Failed - is that correct?

Reply:
Yes. An update that contains errors should cause a rollback, but even when the rollback succeeds the overall update still failed, an I expect stacker to fail too.

Reply:
Added comments to point this out in the declaration site and the stack state handling code.